src/cell_group.hpp

```diff
@@ -231,7 +241,7 @@ class cell_group {
         auto handle = get_probe_handle(probe_id);
 
         auto sampler_index = uint32_t(samplers_.size());
-        samplers_.push_back({handle, s});
+        samplers_.push_back({handle, probe_id.gid, s});
```

Comment:
The cell group needs to know the gid associated with the probe, so it can test the sampler's requested time against the time for that gid. It is a consequence of the asynchronous behaviour.